lib/nimble_totp.ex

```diff
@@ -185,14 +222,51 @@ defmodule NimbleTOTP do
   In this example by validating first against the current time, but also against 30 seconds
   ago, we allow the _previous_ code, to be still valid.
   """
+  @spec valid?(binary(), [option() | validate_option()]) :: boolean()
   def valid?(secret, otp, opts \\ [])
 
   def valid?(secret, <<a1, a2, a3, a4, a5, a6>>, opts) do
     <<e1, e2, e3, e4, e5, e6>> = verification_code(secret, opts)
```

Comment:
What do you think about a `verification_code/3` that receives the time and the period, so we don't need to parse it in `verification_code` and then again inside `reused?`?

Reply:
I changed this and made the new method private, didn't know if you wanted to expose it but thought not. Let me know what you think :)

Reply:
Yeah, private! Perfect!